skopt/wrappers/search_cv.py

```diff
@@ -0,0 +1,490 @@
+from skopt import Optimizer
+from skopt.learning import GaussianProcessRegressor, ExtraTreesRegressor
+
+from sklearn.base import clone
+from sklearn.externals.joblib import Parallel, delayed, cpu_count
+import sklearn.model_selection._search as skms
+
+import numpy as np
+from collections import *
+
+
+class SkoptSearchCV(skms.BaseSearchCV):
+    """Bayesian optimization over hyper parameters.
+
+    SkoptSearchCV implements a "fit" and a "score" method.
+    It also implements "predict", "predict_proba", "decision_function",
+    "transform" and "inverse_transform" if they are implemented in the
+    estimator used.
+
+    The parameters of the estimator used to apply these methods are optimized
+    by cross-validated search over parameter settings.
+
+    In contrast to GridSearchCV, not all parameter values are tried out, but
+    rather a fixed number of parameter settings is sampled from the specified
+    distributions. The number of parameter settings that are tried is
+    given by n_iter.
+
+    Parameters are presented as a list of skopt.space.Dimension objects.
+
+    Parameters
+    ----------
+    estimator : estimator object.
+        A object of that type is instantiated for each grid point.
+        This is assumed to implement the scikit-learn estimator interface.
+        Either estimator needs to provide a ``score`` function,
+        or ``scoring`` must be passed.
+
+    search_spaces : list of dict or tuple
+        Either list  of dictionary objects or list of tuples of
+        (dict, int > 0). The keys of every dictionary are parameter
+        names (strings) and values are skopt.space.Dimension instances
+        (Real, Integer or Categorical) which represents search space
+        for particular parameter.
+        If a list of dictionary objects is given, then the search is
+        performed sequentially for every parameter space with maximum
+        number of evaluations set to self.n_iter. Alternatively, if
+        a list of (dict, int > 0) is given, the search is done for
+        every search space for number of iterations given as a second
+        element of tuple.
```

Comment:
The documentation for `GridSearchCV` says that if the list of dicts is provided, then grid spanned by every dict is explored sequentially. Hence I do the same in here.